### PR TITLE
fix: org:delete scratch org + sandboxes

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
migrates `org:delete` from toolbelt
carries over messages and error messages
uses sfdx-core for new `delete` method on `Org`

requires https://github.com/forcedotcom/sfdx-core/pull/491

### CODE REVIEW

File: src/commands/force/org/delete.ts [hunk 1]
```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as os from 'os';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { Messages, Org } from '@salesforce/core';
+import { SandboxOrgConfig } from '@salesforce/core/lib/config/sandboxOrgConfig';
```
Comment (mshanemc) [inline src/commands/force/org/delete.ts:10]: `SandboxOrgConfig` doesn't seem to be intentionally exposed.

Why not wrap all of `(await this.org.getSandboxOrgConfigField(SandboxOrgConfig.Fields.PROD_ORG_USERNAME))`;
in a new `org.isSandbox()` method?

File: test/commands/force/org/delete.test.ts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+      const orgStub = fromStub(stubInterface<Org>(sandbox, orgStubOptions));
+      cmd.setOrg(orgStub);
+    });
+    uxConfirmStub = stubMethod(sandbox, UX.prototype, 'confirm');
+    uxLogStub = stubMethod(sandbox, UX.prototype, 'log');
+
+    return cmd.runIt();
+  };
+
+  beforeEach(() => {});
```
Comment (shetzel) [inline test/commands/force/org/delete.test.ts:86]: this `beforeEach` can be deleted.

File: src/commands/force/org/delete.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      this.flags.noprompt ||
+      (await this.ux.confirm(messages.getMessage('confirmDelete', [isSandbox ? 'sandbox' : 'scratch', username])))
+    ) {
+      let alreadyDeleted = false;
+      let successMessageKey = 'commandSandboxSuccess';
+      try {
+        // will determine if it's a scratch org or sandbox and will delete from the appropriate parent org (DevHub or Production)
+        await this.org.delete();
+      } catch (e) {
+        if (e instanceof Error && e.name === 'attemptingToDeleteExpiredOrDeleted') {
```
Comment (shetzel) [inline src/commands/force/org/delete.ts:47]: Error names should really be less specific and start with a capital letter.  This looks more like an error message.

**Review by shetzel — APPROVED** [review]
Approve with some nits

**Pull request merged**